monai/data/image_writer.py

```diff
@@ -0,0 +1,395 @@
+            spatial_ndim: modifying the spatial dims if needed, so that output to have at least
+                this number of spatial dims. If ``None``, the output will have the same number of
+                spatial dimensions as the input.
+            contiguous: if ``True``, the output will be contiguous.
+        """
+        # change data to "channel last" format
+        if channel_dim is not None:
+            # _chns = ensure_tuple_rep(channel_dim)
+            # data = moveaxis(data, _chns, tuple(range(-len(_chns), 0)))
+            data = moveaxis(data, channel_dim, -1)
+        else:  # adds a channel dimension
+            data = data[..., None]
+        # To ensure at least three spatial dims
+        if spatial_ndim:
+            while len(data.shape) < spatial_ndim + 1:  # assuming the data has spatial + channel dims
+                data = data[..., None, :]
+            while len(data.shape) > spatial_ndim + 1:
+                data = data[..., 0, :]
+        # if desired, remove trailing singleton dimensions
+        while squeeze_end_dims and data.shape[-1] == 1:
+            data = np.squeeze(data, -1)
+        if contiguous:
+            data = ascontiguousarray(data)
+        return data
+
+    @classmethod
+    def get_meta_info(cls, metadata: Optional[Mapping] = None):
+        """
+        Extracts relevant meta information from the metadata object (using ``.get``).
+
+        This method returns the following fields (the default value is ``None``):
+
+            - ``'original_affine'``: for data original affine (before any image processing),
+            - ``'affine'``: for the current data affine (representing the current coordinate information),
+            - ``'spatial_shape'``: for data original spatial shape.
+        """
+        if not metadata:
+            default_dict = {"original_affine": None, "affine": None, "spatial_shape": None}
+            metadata = default_dict
+        original_affine = metadata.get("original_affine")
+        affine = metadata.get("affine")
+        spatial_shape = metadata.get("spatial_shape")
+        return original_affine, affine, spatial_shape
+
+
+@require_pkg(pkg_name="itk")
+class ITKWriter(ImageWriter):
+    """
+    Write data and metadata into files on disk using ITK-python.
+    """
+
+    def __init__(self, output_dtype: DtypeLike = np.float32, **kwargs):
+        super().__init__(output_dtype=output_dtype, affine=None, channel_dim=0, **kwargs)
+
+    def set_data_array(self, data_array, channel_dim: Optional[int] = 0, squeeze_end_dims: bool = True, **kwargs):
+        """Convert ``data_array`` into 'channel-last' numpy ndarray."""
+        self.data_obj = self.convert_to_channel_last(
+            data=data_array,
+            channel_dim=channel_dim,
+            squeeze_end_dims=squeeze_end_dims,
+            spatial_ndim=kwargs.pop("spatial_ndim", 3),
```

Comment:
I feel it's necessary to describe all the supported args in `kwargs` in the doc-string.

Thanks.